src/Aspire.Hosting.Azure/AzureCosmosDBResource.cs

```diff
@@ -85,6 +146,55 @@ public static IResourceBuilder<AzureCosmosDBResource> AddAzureCosmosDB(this IDis
                       .WithManifestPublishingCallback(resource.WriteToManifest);
     }
 
+    /// <summary>
+    /// Adds an Azure Cosmos DB connection to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="configureResource"></param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureCosmosDBConstructResource> AddAzureCosmosDBConstruct(this IDistributedApplicationBuilder builder, string name, Action<IResourceBuilder<AzureCosmosDBConstructResource>, ResourceModuleConstruct, CosmosDBAccount, IEnumerable<CosmosDBSqlDatabase>>? configureResource = null)
+    {
+        var configureConstruct = (ResourceModuleConstruct construct) =>
+        {
+            var cosmosAccount = new CosmosDBAccount(construct, CosmosDBAccountKind.GlobalDocumentDB, name: name);
+            cosmosAccount.AssignProperty(x => x.ConsistencyPolicy.DefaultConsistencyLevel, "'Session'");
+            cosmosAccount.AssignProperty(x => x.DatabaseAccountOfferType, "'Standard'");
+            cosmosAccount.AssignProperty(x => x.Locations[0].LocationName, "location");
+            cosmosAccount.AssignProperty(x => x.Locations[0].FailoverPriority, "0");
+
+            var keyVaultNameParameter = new Parameter("keyVaultName");
+            construct.AddParameter(keyVaultNameParameter);
+
+            var azureResource = (AzureCosmosDBConstructResource)construct.Resource;
+            var azureResourceBuilder = builder.CreateResourceBuilder(azureResource);
+            List<CosmosDBSqlDatabase> cosmosSqlDatabases = new List<CosmosDBSqlDatabase>();
+            foreach (var databaseName in azureResource.Databases)
+            {
+                var cosmosSqlDatabase = new CosmosDBSqlDatabase(construct, cosmosAccount, name: databaseName);
+                cosmosSqlDatabases.Add(cosmosSqlDatabase);
+            }
+
+            var keyVault = KeyVault.FromExisting(construct, "keyVaultName");
+
+            var keyVaultSecret = new KeyVaultSecret(construct, keyVault, "connectionString");
+            keyVaultSecret.AssignProperty(
+                x => x.Properties.Value,
```

Comment:
Can you use cosmosAccount.GetConnectionString? https://github.com/Azure/azure-sdk-for-net/blob/feature/cdk/sdk/provisioning/Azure.Provisioning/src/cosmosdb/CosmosDBAccount.cs#L81